pkg/scheduler/internal/cache/cache.go

```diff
@@ -477,22 +505,13 @@ func (cache *schedulerCache) AddPod(pod *v1.Pod) error {
 		if currState.pod.Spec.NodeName != pod.Spec.NodeName {
 			// The pod was added to a different node than it was assumed to.
 			klog.InfoS("Pod was added to a different node than it was assumed", "pod", klog.KObj(pod), "assumedNode", klog.KRef("", pod.Spec.NodeName), "currentNode", klog.KRef("", currState.pod.Spec.NodeName))
-			// Clean this up.
-			if err = cache.removePod(currState.pod); err != nil {
-				klog.ErrorS(err, "Error occurred while removing pod")
-			}
-			cache.addPod(pod)
 		}
-		delete(cache.assumedPods, key)
-		cache.podStates[key].deadline = nil
-		cache.podStates[key].pod = pod
+		if err = cache.updatePod(currState.pod, pod, key); err != nil {
+			klog.ErrorS(err, "Error occurred while updating pod")
+		}
 	case !ok:
 		// Pod was expired. We should add it back.
-		cache.addPod(pod)
-		ps := &podState{
-			pod: pod,
-		}
-		cache.podStates[key] = ps
+		cache.addPod(pod, key, false)
```

Comment:
False `isAssumed` indicates we don't care about whether it exits in assumedPods or not, right?

(BTW: I'd suggest looking through all usages of addPod() and check if the value of `isAssumed` is passed in properly)

Reply:
No. More precisely: As a function parameter (in `addPod`), `isAssumed` indicates whether the pod should be added to the `assumedPod`. **This value is `true` if and only if `addPod` is called in `AssumePod`, and the others are `false`.**

Reply:
I see, you want to use this parameter to assume the pod as an optional step. In this case, `assumePod` as the parameter names reads better. (assume as a verb)

Reply:
Absolutely right. And any other questions besides this? If not, I will squash these commits after work ^ ^

Reply:
I will take another look tomorrow.